patch-package
--- a/node_modules/react-scripts/config/webpack.config.dev.js
+++ b/node_modules/react-scripts/config/webpack.config.dev.js
@@ -221,7 +221,7 @@ module.exports = {
           // The preset includes JSX, Flow, and some ESnext features.
           {
             test: /\.(js|mjs|jsx)$/,
-            include: paths.appSrc,
+            include: [paths.appSrc, /metronome-wallet-ui-logic/],
             loader: require.resolve('babel-loader'),
             options: {
               customize: require.resolve(
--- a/node_modules/react-scripts/config/webpack.config.prod.js
+++ b/node_modules/react-scripts/config/webpack.config.prod.js
@@ -296,7 +296,7 @@ module.exports = {
           // The preset includes JSX, Flow, and some ESnext features.
           {
             test: /\.(js|mjs|jsx)$/,
-            include: paths.appSrc,
+            include: [paths.appSrc, /metronome-wallet-ui-logic/],
 
             loader: require.resolve('babel-loader'),
             options: {
--- a/node_modules/react-scripts/scripts/utils/createJestConfig.js
+++ b/node_modules/react-scripts/scripts/utils/createJestConfig.js
@@ -51,7 +51,7 @@ module.exports = (resolve, rootDir, isEjecting) => {
       '^(?!.*\\.(js|jsx|css|json)$)': resolve('config/jest/fileTransform.js'),
     },
     transformIgnorePatterns: [
-      '[/\\\\]node_modules[/\\\\].+\\.(js|jsx)$',
+      '[/\\\\]node_modules[/\\\\](?!(metronome-wallet-ui-logic)/).+\\.(js|jsx)$',
       '^.+\\.module\\.(css|sass|scss)$',
     ],
     moduleNameMapper: {
