patch-package
--- a/node_modules/react-scripts/config/webpack.config.dev.js
+++ b/node_modules/react-scripts/config/webpack.config.dev.js
@@ -221,7 +221,7 @@ module.exports = {
           // The preset includes JSX, Flow, and some ESnext features.
           {
             test: /\.(js|mjs|jsx)$/,
-            include: paths.appSrc,
+            include: [paths.appSrc, /metronome-wallet-ui-logic/],
             loader: require.resolve('babel-loader'),
             options: {
               customize: require.resolve(
--- a/node_modules/react-scripts/config/webpack.config.prod.js
+++ b/node_modules/react-scripts/config/webpack.config.prod.js
@@ -296,7 +296,7 @@ module.exports = {
           // The preset includes JSX, Flow, and some ESnext features.
           {
             test: /\.(js|mjs|jsx)$/,
-            include: paths.appSrc,
+            include: [paths.appSrc, /metronome-wallet-ui-logic/],
 
             loader: require.resolve('babel-loader'),
             options: {
